cranelift/filetests/filetests/runtests/atomic-load-store.clif

```diff
@@ -0,0 +1,105 @@
+test interpret
+test run
+set enable_llvm_abi_extensions=true
+target x86_64
+target aarch64
+target riscv64
+target s390x
+
+function %i64_atomic_store_load(i64) -> i8 {
+    ss0 = explicit_slot 8
+
+block0(v0: i64):
+    v1 = stack_addr.i64 ss0
+    atomic_store.i64 v0, v1
+    v2 = atomic_load.i64 v1
+
+    v3 = icmp.i64 eq v0, v2
+    return v3
+}
+; run: %i64_atomic_store_load(0) == 1
+; run: %i64_atomic_store_load(-1) == 1
+; run: %i64_atomic_store_load(0x00000000_FFFFFFFF) == 1
+; run: %i64_atomic_store_load(0xFFFFFFFF_00000000) == 1
+; run: %i64_atomic_store_load(0xFEDCBA98_76543210) == 1
+; run: %i64_atomic_store_load(0xA00A00A0_0A00A00A) == 1
+; run: %i64_atomic_store_load(0xC0FFEEEE_DECAFFFF) == 1
+
+
+function %i32_atomic_store_load(i32) -> i8 {
+    ss0 = explicit_slot 4
+
+block0(v0: i32):
+    v1 = stack_addr.i64 ss0
+    atomic_store.i32 v0, v1
+    v2 = atomic_load.i32 v1
+
+    v3 = icmp.i32 eq v0, v2
+    return v3
+}
+; run: %i32_atomic_store_load(0) == 1
+; run: %i32_atomic_store_load(-1) == 1
+; run: %i32_atomic_store_load(0x0000FFFF) == 1
+; run: %i32_atomic_store_load(0xFFFF0000) == 1
+; run: %i32_atomic_store_load(0xFEDC3210) == 1
+; run: %i32_atomic_store_load(0xA00AA00A) == 1
+; run: %i32_atomic_store_load(0xC0FFEEEE) == 1
+
+
+function %i16_atomic_store_load(i16) -> i8 {
+    ss0 = explicit_slot 2
+
+block0(v0: i16):
+    v1 = stack_addr.i64 ss0
+    atomic_store.i16 v0, v1
+    v2 = atomic_load.i16 v1
+
+    v3 = icmp.i16 eq v0, v2
+    return v3
+}
+; run: %i16_atomic_store_load(0) == 1
+; run: %i16_atomic_store_load(-1) == 1
+; run: %i16_atomic_store_load(0x00FF) == 1
+; run: %i16_atomic_store_load(0xFF00) == 1
+; run: %i16_atomic_store_load(0xFE10) == 1
+; run: %i16_atomic_store_load(0xA00A) == 1
+; run: %i16_atomic_store_load(0xC0FF) == 1
+
+
+function %i8_atomic_store_load(i8) -> i8 {
+    ss0 = explicit_slot 1
+
+block0(v0: i8):
+    v1 = stack_addr.i64 ss0
+    atomic_store.i8 v0, v1
+    v2 = atomic_load.i8 v1
+
+    v3 = icmp.i8 eq v0, v2
+    return v3
+}
+; run: %i8_atomic_store_load(0) == 1
+; run: %i8_atomic_store_load(-1) == 1
+; run: %i8_atomic_store_load(0x0F) == 1
+; run: %i8_atomic_store_load(0xF0) == 1
+; run: %i8_atomic_store_load(0xAA) == 1
+; run: %i8_atomic_store_load(0xC0) == 1
+
+
+function %atomic_store_load_aligned(i64) -> i8 {
+    ss0 = explicit_slot 16
+
+block0(v0: i64):
+    v1 = stack_addr.i64 ss0
+    atomic_store.i64 aligned v0, v1
+    v2 = atomic_load.i64 v1
```

Comment:
If we're setting the `aligned` flag on the store, should we set it on the load too?